superduperdb/db/base/backends.py

```diff
@@ -7,23 +7,21 @@
 from superduperdb.db.mongodb.data_backend import MongoDataBackend
 from superduperdb.db.mongodb.metadata import MongoMetaDataStore
 from superduperdb.db.sqlalchemy.metadata import SQLAlchemyMetadata
-from superduperdb.vector_search.inmemory import InMemoryVectorDatabase
-from superduperdb.vector_search.lance import LanceVectorDatabase
+from superduperdb.vector_search.lance import LanceVectorIndex
+from superduperdb.vector_search.in_memory import InMemoryVectorIndex
 
 data_backends = {'mongodb': MongoDataBackend, 'ibis': IbisDataBackend}
 
 artifact_stores = {'mongodb': MongoArtifactStore, 'filesystem': FileSystemArtifactStore}
 
 metadata_stores = {'mongodb': MongoMetaDataStore, 'sqlalchemy': SQLAlchemyMetadata}
 
-vector_data_stores = {
-    'lance': LanceVectorDatabase,
-    'inmemory': InMemoryVectorDatabase,
+vector_index_impl = {
```

Comment:
are we sure this is good name?

Reply:
I can't think of a better name. We could call it just "numpy" vector-searcher.

Reply:
Suggestions welcome.